packages/react-native/ReactCommon/react/nativemodule/core/platform/android/ReactCommon/JavaTurboModule.cpp

```diff
@@ -468,7 +468,39 @@ jsi::Value JavaTurboModule::invokeJavaMethod(
       } else {
         TMPL::asyncMethodCallFail(moduleName, methodName);
       }
-      throw;
+      auto exception = std::current_exception();
+      auto throwable = facebook::jni::getJavaExceptionForCppException(exception);
```

Comment:
Extract this to a helper method? `createJSCauseForException`?

Reply:
Yes, would `convertThrowableToJSError` make sense, too?